pkg/networkservice/common/mechanisms/server.go

```diff
@@ -55,22 +54,25 @@ func NewServer(mechanisms map[string]networkservice.NetworkServiceServer) networ
 func (m *mechanismsServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
 	if request.GetConnection().GetMechanism() != nil {
 		srv, ok := m.mechanisms[request.GetConnection().GetMechanism().GetType()]
-		if ok {
-			conn, err := srv.Request(ctx, request)
-			if err != nil {
-				return nil, err
-			}
-			request.Connection = conn
-			return next.Server(ctx).Request(ctx, request)
+		if !ok {
+			return nil, errors.Errorf("Unsupported Mechanism: %+v", request.GetConnection().GetMechanism())
 		}
-		return nil, errors.Errorf("Unsupported Mechanism: %+v", request.GetConnection().GetMechanism())
+		conn, err := srv.Request(ctx, request)
+		if err == nil {
+			return conn, nil
+		}
+		// if err != nil try any available mechanism below
 	}
 	for _, mechanism := range request.GetMechanismPreferences() {
```

Comment:
Let's ignore already failed mechanism type.

Reply:
I think we also need to wrap errors and if no one mechanism server did not return positive answer then need to return this wrapped error :)

Reply:
@denis-tingajkin What are you hoping to get from wrapping here?  I ask because double or triple wrapping can lead to multiple stack traces making it hard to follow.  Say if the error occured at:

Frame1
Frame2
...
Frame20
and was wrapped or withstacked traced... and then at Frame13, we wrap it again, and at Frame2 we wrap it again, then you will get when you %+v it:

ErrMsgFrame2
Frame2
Frame1
ErrMsgFrame13
Frame13
Frame12
...
Frame2
Frame1
ErrMsgFrame20
Frame20
Frame19
...
Frame 13
...
Frame2
Frame1

it gets super super hard to see what's happening... generally wrapping once is usually sufficient, as close to origin of the error from a non-wrapping regime as possible.

Reply:
@edwarnicke makes sense.

I think then we can simply log each negative answer via `log.Error(err.Error())`